admin-api-server/src/analytics/service/analytics.service.ts

```diff
@@ -241,22 +258,37 @@ LIMIT ${params.pageSize}
     }
 
     return {
-      data: qryRes.rows.map(
-        (row: any) =>
-          <Activity>{
-            id: Number(row['id']),
-            timestamp: Math.floor(row['timestamp'].getTime() / 1000),
-            kind: row['kind'],
-            from: row['from'],
-            to: row['to'],
-            tokenId: Number(row['tokenId']),
-            price: this.currencyService.convertToCurrency(
-              Number(row['price']),
-              BASE_CURRENCY,
-            ),
-            amount: Number(row['amount']),
-          },
-      ),
+      data: qryRes.rows.map((row: any) => {
+        const price = this.currencyService.convertToCurrency(
+          Number(row['price']),
+          BASE_CURRENCY,
+        );
+
+        const nftPriceSumBase = this.currencyService.convertToCurrency(
+          row['nft_price_sum'],
+          BASE_CURRENCY,
+        );
+        const conversionRate =
+          row['nft_order_price'] &&
+          Number(nftPriceSumBase) / row['nft_order_price'];
+
+        return <Activity>{
+          id: Number(row['id']),
+          timestamp: Math.floor(row['timestamp'].getTime() / 1000),
+          kind: row['kind'],
+          from: row['from'],
+          to: row['to'],
+          tokenId: Number(row['tokenId']),
+          price,
+          edition_size: Number(row['edition_size']),
+          currency: row['currency'],
+          transaction_value:
+            price &&
```

Comment:
No need to change this here, this is more a comment in general.

This && construct, I am not sure this is nice. Syntactically it reads nice, but semantically doesn't this mean that if price is undefined transaction value becomes `false`? Which is weird, it would be more correct for transaction_value to become `undefined` or `null` in this scenario. 

I think therefore it makes more sense to do it this way `price && conversionRate ? (Number(price) / conversionRate).toFixed(2) : undefined`.

Reply:
This can only become false if either price or conversionRate are false. Here is an example: https://www.typescriptlang.org/play?#code/MYewdgzgLgBADgJwJbAKYwLwwK5gCaoBmSYqeA3AFCWiSwTYJgCGAtulgEQQjsvucqNcDwA2qAHSiQAcwAUiFOgBkymAyZsVagAZQAFkggwjMcKICe6-SADuYGLf2oHitDGb51jfumYJ0KARsVAAuGAASAG83VABfHQBKIA

Reply:
Aaha nice. Then I am happy after all haha.

I also tried `null`, and the result becomes `null`. That too is sensible.